File: oojs-orange-tree/src/orange_tree.js
var OrangeTree = function(age, height, oranges) {
  this.age = age;
  this.height = height;
  this.oranges = oranges;
}

OrangeTree.prototype.isMature = function(){
  if (this.age >= 6){
    return true;
  } else {
    return false;
  }
}

OrangeTree.prototype.isDead = function(){
  if (this.age === 100) {
    return true;
  } else {
    return false;
  }

}

OrangeTree.prototype.hasOranges = function(){
  if (this.oranges.length > 0){
    return true;
  } else {
    return false;
  }
}

OrangeTree.prototype.maximumHeight = 25;

OrangeTree.prototype.passGrowingSeason = function(){
  this.oranges = [];
  if (!this.isDead()){
    this.age ++;
    this.height += 2.5;
    if (this.height > 25){
      this.height = 25;
    }
    if (this.isMature()){
      for (i = 0; i < 10; i++){
        this.oranges.push(new Orange());
      }
    }
  }
}

OrangeTree.prototype.pickAnOrange = function(){
    // console.log("hello");
    // console.log(this.oranges.pop());
    return this.oranges.pop();
}

File: orange-groves/orange_tree.rb
require_relative 'fruit_tree'

class OrangeTree < FruitTree
  class NoOrangesError < StandardError
  end

  MAXIMUM_AGE = 100
  MAXIMUM_HEIGHT = 25
  MATURE_AGE = 6
  ANNUAL_GROWTH = 2.5
  YEAR = 1

  attr_accessor :oranges

  def initialize
    super
    @oranges = []
  end

  def pass_growing_season
    self.oranges.clear
    self.age += YEAR
    if height < MAXIMUM_HEIGHT
      self.height += ANNUAL_GROWTH
    end
    if mature?
      self.oranges = Array.new(rand(100..300)) { Orange.new }
    end
  end

  def mature?
    age >= MATURE_AGE
  end

  def dead?
    age >= MAXIMUM_AGE
  end

  def has_oranges?
    oranges.length > 0
  end

  def pick_an_orange
    raise NoOrangesError, "This tree has no oranges" unless self.has_oranges?

    self.oranges.pop
  end
end




